cirq/sim/density_matrix_utils_test.py

```diff
@@ -27,12 +27,12 @@ def assert_valid_density_matrix(matrix, num_qubits=1):
 
 
 def test_von_neumann_entropy():
-    with pytest.raises(ValueError, match='zero'):
-        cirq.von_neumann_entropy(np.array([[1, 0], [0, 0]]))
+    assert np.isclose(cirq.von_neumann_entropy(np.array([[0.5, 0], [0, 0.5]])),
+                      -np.log(2), 1e-02)
 
     assert np.isclose(
-        cirq.von_neumann_entropy(np.array([[0.5, 0.5j], [-0.5j, 0.5]])),
-        -0.6931471805599453 + 0j)
+        cirq.von_neumann_entropy(np.array(np.array([[0.5, 0.5], [0.5, 0.5]]))),
```

Comment:
Test the following things:

- The 1x1 matrix [[1]].

- A 2x2 density matrix created by calling random unitary, using each column as a ket, turning each ket into a density matrix, and then doing a linear combination of the two with coefficients 0.1 and 0.9. The entropy should be `-0.1 log_2(0.1) - 0.9 log_2(0.9) ~= 0.46899559358`

- An EPR pair state (|00> + |11>)(<00| + <11|). (entropy 0)

Some sort of 3x3 state.

Some sort of 4x4 state.

Reply:
> An EPR pair state (|00> + |11>)(<00| + <11|)

I think i'm doing this right. With this i get some eigenvalues which are 0, which i can't take the log of.

```python
density_matrix = np.array([1, 0, 0, 1] * np.array([[1], [0], [0], [1]]))
eigenvalues = np.linalg.eigvals(density_matrix)
np.log2(eigenvalues) # RuntimeWarning: divide by zero encountered in log2 np.log2(eigenvalues)
```